control/controlclient/direct.go

```diff
@@ -1602,6 +1602,57 @@ func postPingResult(start time.Time, logf logger.Logf, c *http.Client, pr *tailc
 	return nil
 }
 
+type ClientAuditLog struct {
+	// The action must correspond to a known action in the control plane.
+	Action string
+	// The details will be use to populate ActionDetails  in the audit
+	// log request.
+	Details string
+}
+
+// RequestAuditLog sends an audit log to the control plane.
+//
+// Clients are responsible for checking the error response and delaying
+// any actions as required until a successful response is received.
+// Clients are responsible for retries as requires.
+//
+// returns and error and a bool indicating if the error is transient
+// and the request can be retried.
+func (c *Direct) SendAuditLog(ctx context.Context, auditLog ClientAuditLog) (error, bool) {
```

Comment:
Maybe something like this?

```suggestion
// It returns an error and a bool indicating whether the error is transient
// and the request can be retried.
func (c *Direct) SendAuditLog(ctx context.Context, auditLog ClientAuditLog) (err error, retry bool) {
```